src/azure-cli/azure/cli/command_modules/acr/custom.py

```diff
@@ -245,3 +260,45 @@ def _check_wincred(login_server):
         return True
 
     return False
+
+
+def _create_diagnostic_settings(cli_ctx, acr, workspace):
+    from azure.mgmt.monitor import MonitorManagementClient
+    from azure.mgmt.monitor.models import DiagnosticSettingsResource
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    client = get_mgmt_service_client(cli_ctx, MonitorManagementClient)
+    logs = [
+        {
+            "category": "ContainerRegistryRepositoryEvents",
+            "enabled": True,
+            "retentionPolicy": {
+                "days": 0,
+                "enabled": False
+            }
+        },
+        {
+            "category": "ContainerRegistryLoginEvents",
+            "enabled": True,
+            "retentionPolicy": {
+                "days": 0,
+                "enabled": False
+            }
+        }
+    ]
+    metrics = [
+        {
+            "category": "AllMetrics",
+            "enabled": True,
+            "retentionPolicy": {
+                "days": 0,
+                "enabled": False
+            },
+            "timeGrain": None
+        }
+    ]
```

Comment:
Actually, using json/dict is a more favored style from Python sdk's aspect because of the simplicity and good readabilities; however since the surrounding code uses model classes, be consistent with them make more sense. Let me update it.

Reply:
By design, Azure CLI should call Python SDK. But in fact, you can use JSON directly. Being consistent with the existing code in your module is always a good choice.